react/content-list.jsx

```diff
@@ -45,14 +58,30 @@ let ContentList = React.createClass({
     return this.setStateFromDattCore()
   },
 
+  componentWillUnmount: function () {
+    this.unmonitorDattCore()
+  },
+
   componentWillReceiveProps: function () {
     return this.setStateFromDattCore()
   },
 
   monitorDattCore: function () {
     let dattcore = this.props.dattcore
+    let uiEvents = this.props.uiEvents
+
     dattcore.on('peers-content-auth', this.handlePeersContentAuth)
     dattcore.on('content-content-auth', this.handleContentContentAuth)
+    uiEvents.on('refresh-content', this.handleRefreshContent)
```

Comment:
Much better :) But also for the record, as you mentioned somewhere else, we should probably use Flux - people complain about creating a mess by using events+react, so we should probably just adopt Flux.